cmd/readiness/main.go

```diff
@@ -73,33 +69,35 @@ func isPodReady(conf config.Config) (bool, error) {
 	}
 
 	if inGoalState && inReadyState {
-		logger.Info("Agent has reached goal state")
+		logger.Info("The Agent has reached goal state. Returning ready.")
 		return true, nil
 	}
 
 	// Fallback logic: the agent is not in goal state and got stuck in some steps
-	if !inGoalState && hasDeadlockedSteps(healthStatus) {
+	if !inGoalState && isOnWaitingStep(healthStatus) {
+		logger.Info("The Agent is on wait Step. Returning ready.")
 		return true, nil
 	}
 
+	logger.Info("Reached the end of the check. Returning not ready.")
```

Comment:
A note for the future - we might want to consider building a logging context with debug information and once e return not ready, we could append why. Potentially this could be implemented withing this or next ticket.

Reply:
On the second thought - we return not ready when we don't fall into any other condition in the checking loop. The only thing we could log there is - 👆 statement. And this doesn't make sense.